metals/src/main/scala/scala/meta/internal/metals/NewFilesProvider.scala

```diff
@@ -102,9 +106,12 @@ class NewFilesProvider(
     val path = directory.getOrElse(workspace).resolve(name + ".scala")
     //name can be actually be "foo/Name", where "foo" is a folder to create
     val className = directory.getOrElse(workspace).resolve(name).filename
+    val template = kind match {
+      case caseClassPick.id => caseClassTemplate(name)
```

Comment:
You should use `className` here and below.

Reply:
Or course, thanks for spotting the bug! I’ll fix it soon